rita_client/src/operator_update/tests.rs

```diff
@@ -0,0 +1,260 @@
+            current_hour,
+        )
+        .unwrap();
+        let res_usage_client = res_usage.client_bandwidth;
+        let res_usage_relay = res_usage.relay_bandwidth;
+
+        // check that the sorting in process_usage_data is correct after shuffling and sending it through
+        assert!(
+            res_usage_relay.get(0).unwrap().index < res_usage_relay.get(1).unwrap().index
+                && res_usage_relay.get(1).unwrap().index < res_usage_relay.get(2).unwrap().index
+        );
+        assert!(
+            res_usage_client.get(0).unwrap().index < res_usage_client.get(1).unwrap().index
+                && res_usage_client.get(1).unwrap().index < res_usage_client.get(2).unwrap().index
+        );
+        // check that the binary searching is correct: we did not remove any entries from usage client; so we should have started exactly
+        // from the last seen hour. we removed the last seen from usage relay, so we should expect to see our earliest hour as one fewer
+        assert!(res_usage_client.get(0).unwrap().index == last_seen_hour);
+        assert!(res_usage_relay.get(0).unwrap().index == last_seen_hour);
+
+        // now check that same thing, but in case we have a gap in the data. we'll remove the entry for the last_seen_hour from usage_data_relay
+        // to make sure we are successfully returning the next earliest hour (in our case, the last seen -1). we use res_usage client and relay
+        // because to remove the correct entry it needs to be presorted (if we've gotten here it's guaranteed.)
+        // we successfully search for the entry or return the next one down.
+        unshuffled_client.remove(unshuffled_client.len() - 11);
+        unshuffled_relay.remove(unshuffled_relay.len() - 11);
+        // so the index of our last seen hour if we say last seen is current - 10... will be at len - 11.
+        let res_usage = process_usage_data(
+            unshuffled_client,
+            unshuffled_relay,
+            last_seen_hour,
+            current_hour,
+        )
+        .unwrap();
+        let res_usage_client = res_usage.client_bandwidth;
+        let res_usage_relay = res_usage.relay_bandwidth;
+        // after processing we should start at last seen - 1.
+        println!(
+            "{:?} last seen {:?}",
+            res_usage_relay.get(0).unwrap().index,
+            last_seen_hour
+        );
+        assert!(res_usage_relay.get(0).unwrap().index == last_seen_hour - 1);
+        assert!(res_usage_client.get(0).unwrap().index == last_seen_hour - 1);
+
+        // check that our iteration function does indeed stop at a month of data:
+        assert!(res_usage_client.len() <= 730);
+        assert!(res_usage_relay.len() <= 730);
+    }
+
+    // generates a usage tracker struct for testing without payments since these do not get sent up in ops updates.
+    // using this while I can't get the import working... as a note the original function needs to be updated to push to back
```

Comment:
this comment may need to be updated? It doesn't look like you ended up needing to duplicate the function.

Reply:
also please generate with payments if you can, you never know when we might start having those here.

Reply:
revised the rita common usage tracker function to be public to operator update in rita client